agnpy/radiative_process/tests/test_time_evolution.py

```diff
@@ -0,0 +1,301 @@
+        TimeEvolution(blob2, time, synchrotron_loss(synch2)).eval_with_fixed_intervals(intervals_count=steps_heun, method="Heun")
+        heun_reversed_analytically = gamma_before_time(synch2._electron_energy_loss_formula_prefactor, blob2.gamma_e, time)
+
+        errors_heun = np.abs((heun_reversed_analytically - initial_gamma) / initial_gamma)
+        errors_euler = np.abs((euler_reversed_analytically - initial_gamma) / initial_gamma)
+        average_error_euler = np.average(errors_euler)
+        average_error_heun = np.average(errors_heun)
+        print("Average error Euler method", average_error_euler)
+        print("Average error Heun  method", average_error_heun)
+        assert average_error_heun < average_error_euler
+        assert np.alltrue(errors_heun <= errors_euler)
+
+    def test_heun_method_compared_to_euler_for_automatic_intervals_method(self):
+        """ Heun method should give better results than Euler method.
+            Interestingly, using Heun instead of Euler method, gives lower errors of the final result
+            even when the step threshold is 2 orders of magnitude less restrictive
+        """
+        time = 60 * u.s
+        precision_euler = 0.00001
+        precision_heun  = 0.001
+
+        blob1 = Blob(n_e=PowerLaw())
+        initial_gamma = blob1.gamma_e
+        synch1 = Synchrotron(blob1)
+        TimeEvolution(blob1, time, synchrotron_loss(synch1)).eval_with_automatic_intervals(max_change_per_interval=precision_euler,
+                                                           method="Euler")
+        euler_reversed_analytically = gamma_before_time(synch1._electron_energy_loss_formula_prefactor,
+                                                        blob1.gamma_e, time)
+
+        blob2 = Blob(n_e=PowerLaw())
+        synch2 = Synchrotron(blob2)
+        TimeEvolution(blob2, time, synchrotron_loss(synch2)).eval_with_automatic_intervals(max_change_per_interval=precision_heun,
+                                                           method="Heun")
+        heun_reversed_analytically = gamma_before_time(synch2._electron_energy_loss_formula_prefactor,
+                                                       blob2.gamma_e, time)
+
+        errors_euler = np.abs((euler_reversed_analytically - initial_gamma) / initial_gamma)
+        errors_heun = np.abs((heun_reversed_analytically - initial_gamma) / initial_gamma)
+        average_error_euler = np.average(errors_euler)
+        average_error_heun = np.average(errors_heun)
+        print("Average error Euler method", average_error_euler)
+        print("Average error Heun  method", average_error_heun)
+        assert average_error_heun < average_error_euler
+        assert np.alltrue(errors_heun <= errors_euler)
+
+    def test_automatic_and_fixed_intervals_approach_with_single_loop(self):
+        """ Automatic intervals approach with threshold low enough should use just one run of calculations, so should give the same result
```

Comment:
shouldn't it be: "with threshold *high* enough" 
the actual threshold is rather low 0.01, but it all depends on the blob parameters

Reply:
fixed